api/v1/openapi.yaml

```diff
@@ -1407,9 +1410,20 @@ definitions:
       deviceMTU:
         description: MTU on workload facing devices
         type: integer
+      deviceIfIndex:
```

Comment:
The above `deviceIfIndex` is basically only internal information, meaning the CNI queries the Cilium daemon for the specified ipvlan master device in order to setup ipvlan slaves bound to it. We could rename the above, or are you saying that the above property name is okay, just that the daemon command line parameter should be renamed to something as specific as `--ipvlan-master-device-index`? I actually liked reusing the `--device` param, so that it generically becomes `--datapath-mode ipvlan --device eth0`. Even if we get rid of `--device`, wouldn't we still have the same issue for ipvlan at that point? Meaning, if we have some generic master device detection in place, wouldn't this be needed for both datapath flavors?

Reply:
The above change to `openapi.yaml` becomes stable API. We can never change it again. If it refers to the IPVLAN master ifindex then let's label it as such.

The main difference between prior use of `--device` that it referred to an ethernet device name which can be different on each worker node depending on HW or node configuration whereas the IPVLAN master device is likely more stable with something such as `ipvlan0`.